ludwig/trainers/trainer.py

```diff
@@ -353,115 +365,78 @@ def train_for_tuning(
-                batcher.set_epoch(epoch, self.batch_size)
-                self.model.reset_metrics()
-                while not batcher.last_batch() and step_count < total_training_steps:
-                    batch = batcher.next_batch()
-                    inputs = {
-                        i_feat.feature_name: torch.from_numpy(np.array(batch[i_feat.proc_column], copy=True)).to(
-                            self.device
-                        )
-                        for i_feat in self.model.input_features.values()
-                    }
-                    targets = {
-                        o_feat.feature_name: torch.from_numpy(np.array(batch[o_feat.proc_column], copy=True)).to(
-                            self.device
-                        )
-                        for o_feat in self.model.output_features.values()
-                    }
-
-                    loss, _ = self.train_step(
-                        inputs,
-                        targets,
-                    )
-                    # compute smoothed loss
-                    avg_loss = beta * avg_loss + (1 - beta) * loss
-                    smoothed_loss = avg_loss / (1 - beta ** (step_count + 1))
-
-                    # store learning rate and loss
-                    learning_rates.append(current_learning_rate)
-                    losses.append(smoothed_loss.detach().cpu().numpy())
-                    logger.info(f"Explored learning_rate={current_learning_rate} loss={smoothed_loss}")
-
-                    # check whether loss is diverging
-                    if step_count > 0 and smoothed_loss > early_stop_threshold * best_loss:
-                        diverging = True
-                        break
-                    else:
-                        if smoothed_loss < best_loss or step_count == 0:
-                            best_loss = smoothed_loss
-
-                    # compute new learning rate
-                    if mode == "exponential":
-                        current_learning_rate = exponential_scheduler(current_learning_rate, step_count)
-                    else:
-                        current_learning_rate = linear_scheduler(current_learning_rate, step_count)
-
-                    self.set_optimizer_learning_rate(current_learning_rate)
-                    step_count += 1
+    def tune_learning_rate(self, config: ModelConfigDict) -> float:
+        """Recommend a learning rate based on the feautres in the configs and the associated encoders.
 
-                epoch += 1
+        param:
+            config: Ludwig config used to train the model.
+        """
+        logger.info("Tuning learning rate...")
 
-        optimal_lr = get_optimal_lr(losses, learning_rates)
-        if optimal_lr:
-            learning_rate = optimal_lr
+        features_with_large_encoders = {TEXT, IMAGE, SEQUENCE, AUTO, TIMESERIES}
+        huggingface_encoders = {
```

Comment:
I worry this list is going to get out of sync very quickly as we add new pretrained encoders. Maybe what we can do instead is have a set of `pretrained_encoders` in both `text_encoders.py` and `image_encoders.py` and then merge them here. That way at least it's all in one place and easier to update when a new encoder gets added.